### DISCUSSION TITLE
Problem using MS SQL server

### DISCUSSION THREAD

**Discussion by jpstrube** [keycloak/keycloak#29307] [Q&A] [answered]
Hi,

I'm trying to start Keyclock from Docker with an external MS SQL Server 2022 database, but I get this error:

`ERROR [org.keycloak.quarkus.runtime.cli.ExecutionExceptionHandler] (main) Error details:: java.lang.RuntimeException: Cannot use default schema name keycloaktest on Microsoft SQL Server because the login schema of the current user (dbo) is different and MSSQL does not support setting the default schema per session.`

When I manually create the configured schema and set it to the user's default schema I get another error:

`ERROR [org.keycloak.quarkus.runtime.cli.ExecutionExceptionHandler] (main) Error details:: org.hibernate.exception.SQLGrammarException: JDBC exception executing SQL [select top (?) mme1_0.ID,mme1_0.UPDATE_TIME,mme1_0.VERSION from keycloaktest.MIGRATION_MODEL mme1_0 order by mme1_0.UPDATE_TIME desc] [Invalid object name 'keycloaktest.MIGRATION_MODEL'.] [n/a]`

So, if the schema does not exist, I cannot connect because it cannot be the user's default schema. And if the schema exists, it seems that all the tables are also expected to exist. That looks to me like a chicken-and-egg problem. How can I get around this?

Thanks,
Jan

**Comment by sschu** [comment, marked as answer]
Are you mixing up schema and database name? We use a custom name for the database but keep dbo as schema name and this works without problems.

**Comment by jpstrube** [comment]
Thanks a lot @sschu, that works.
I did not mix them up but I expected it to work like with MySQL where you can define a custom schema name. Seems like MS SQL is different here.
